docs/DeveloperGuide.md

```diff
@@ -6,8 +6,77 @@
 
 ## Design & implementation
 
+# High Level Functionalities
+![HighLevel.png](HighLevel.png)
+
+**FitTrack** is the main class and entry point of the application.  
+It manages high level functionalities by coordinating the `FitTrackLogger`, `Storage`, `User`, `Parser`, `Ui`, `TrainingSession`, and `Exercise` classes.
+
+- **FitTrackLogger** manages logging for the application, ensuring errors and important events are properly recorded.
+- **Storage** manages saving and loading data from a persistent storage file.
+- **User** holds the user’s information, such as age and gender, and provides methods to modify or retrieve this data.
+- **Parser** handles parsing of user input, converting it into commands and actions.
+- **Ui** manages all output and user interaction, such as printing data and messages to the console.
+- **TrainingSession** represents a single training session, including exercises and metadata (like date and description).
+- **Exercise** represents different types of exercises available in the application, like pull-ups or shuttle runs.
+
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Edit Exercise Feature
+
+The **Edit Exercise** feature is managed by the `TrainingSession` class, and is primarily carried out by its 
+`editExercise()` function. This feature utilizes the `setPerformance()`and `getReps()` methods from the 
+`ExerciseStation` classes to edit the repetitions and timings for the user’s selected 
+exercises. Additionally, it calculates the points the user will earn for each exercise based on the updated values.
+
+#### Step 1: Logging a New Training Session
+
+When the user logs a new training session, an instance of the `TrainingSession` class is created. This instance 
+initializes an EnumMap, which populates the various `ExerciseStation` instances with their initial values. 
+Below is a representation of how the `ExerciseStations` are initialized:
+
+![TrainingSessionInitialState.png](TrainingSessionInitialState.png)  
```

Comment:
Unclear what the visibility is in this class diagram
![image](https://github.com/user-attachments/assets/073c9b53-b9c0-4456-82d4-ba69a972b2bd)